analyzer/block/block.go

```diff
@@ -192,20 +192,9 @@ func (b *blockBasedAnalyzer) Start(ctx context.Context) {
 		}
 		batchCtx, batchCtxCancel = context.WithTimeout(ctx, lockExpiryMinutes*time.Minute)
 
-		var to uint64
-		// Get the latest available block on the source.
-		latestBlockHeight, err := b.processor.SourceLatestBlockHeight(ctx)
-		if err != nil {
-			b.logger.Error("failed to query latest block height on source",
-				"err", err,
-			)
-			backoff.Failure()
-			continue
-		}
-		to = latestBlockHeight
-
+		var to uint64 = util.MaxBlockHeight
```

Comment:
Per discussion from slack with mitja; 

When the analyzers are running in fast sync mode, we expect to run them only to some height h < current_height anyway. Then in slow sync mode there's only one analyzer, so it can lock all it wants -- it will just discover early on in the batch that the block(s) don't exist, it will unlock the rest of the blocks, and go again.

One (expected) side effect of this is that the backoff failure is now triggered repeatedly when the indexer is synced. See `e2e_test.go` for more

Reply:
> One (expected) side effect of this is that the backoff failure is now triggered repeatedly when the indexer is synced. See e2e_test.go for more

I think it would be great if this could be avoided, otherwise this will polute the error log and also unnecessarly increase the backoff.

There used to be this `ErrOutOfRange` (which is still in code, but not actually used anywhere since #389). Maybe it can be introduced back, and in case the block-producer returns it, the block analyzer doesn't log the error (but a debug/info log instead).

Reply:
> I think it would be great if this could be avoided, otherwise this will unnecessarily pollute the error log.

Oh, by _"this"_ i only ment logging the error, should have worded it better. All clear on wanting to have cached calls only.

---
 
Also please move the configuring of the `to` value out of the loop, one level higher, eg. to line 167.

Reply:
Got it; reintroduced `ErrOutOfRange` and moved `to` out

Reply:
Thanks!